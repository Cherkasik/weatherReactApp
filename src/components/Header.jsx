import React from 'react';
import styled from 'styled-components';
import { connect } from 'react-redux';
import { getGeolocation } from '../geolocation/actions';

const Container = styled.div`
  display: flex;
  width: 100%;
  margin-bottom: 20px;
`;

const Text = styled.div`
  font-size: 30pt;
  font-weight: 500;
  color: #85898C;
`;

const RefreshButton = styled.button`
  background-color: #C9D2D7;
  color: #869197;
  border: none;
  border-radius: 5px;
  margin-left: calc(50% - 350px);
  padding: 15px 45px;
  font-weight: 600;
`;


function Header(props) {
  const { refresh } = props;
  return (
    <Container>
      <Text>Погода здесь</Text>
      <RefreshButton onClick={refresh}>Обновить геолокацию</RefreshButton>
    </Container>
  );
}

const mapStateToProps = () => ({});

const mapDispatchToProps = {
  getGeolocation,
};

export default connect(mapStateToProps, mapDispatchToProps)(Header);
