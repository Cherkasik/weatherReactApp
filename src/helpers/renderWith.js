import React from 'react';
import { createStore } from 'redux';
import { Provider } from 'react-redux';
import { render } from '@testing-library/react';
import reducer from '../store/rootReducer';

export function renderWithRedux(
  ui,
  { initialState, store = createStore(reducer, initialState) } = {}
) {
  return {
    ...render(<Provider store={store}>{ui}</Provider>),
    store,
  }
}
