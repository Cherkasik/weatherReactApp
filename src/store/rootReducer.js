import { combineReducers } from 'redux';
import geolocation from '../geolocation/reducer';
import favourites from '../favourites/reducer';
import weather from '../weather/reducer';

const appReducer = combineReducers({
  geolocation,
  favourites,
  weather,
});

const rootReducer = (state, actionValue) => {
  return appReducer(state, actionValue);
};

export default rootReducer;