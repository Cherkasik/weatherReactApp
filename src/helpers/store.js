import thunkMiddleware from "redux-thunk";
import {applyMiddleware, compose, createStore} from "redux";
import rootReducer from "../store/rootReducer";

const middlewares = [thunkMiddleware];

if (process.env.NODE_ENV === 'development') {
  const { logger } = require('redux-logger');
  middlewares.push(logger);
}

export const store = compose(applyMiddleware(...middlewares))(createStore)(rootReducer);