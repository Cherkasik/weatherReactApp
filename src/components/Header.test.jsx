import React from 'react';
import 'jest-styled-components';
import { render } from '@testing-library/react';
import { Provider } from 'react-redux';
import Header from './Header';
import { store } from '../helpers/store';

describe('Header component', () => {
  const refresh = jest.fn();
  it('should render correctly', () => {
    const { container } = render(
      <Provider store={store}>
        <Header refresh={refresh} />
      </Provider>
    );
    expect(container.firstChild).toMatchSnapshot();
  });
});