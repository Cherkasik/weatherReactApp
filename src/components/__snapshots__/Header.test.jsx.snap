// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header component should render correctly 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  width: 100%;
  margin-bottom: 20px;
}

.c1 {
  font-size: 30pt;
  font-weight: 500;
  color: #85898C;
}

.c2 {
  background-color: #C9D2D7;
  color: #869197;
  border: none;
  border-radius: 5px;
  margin-left: calc(50% - 350px);
  padding: 15px 45px;
  font-weight: 600;
}

<div
  class="c0"
>
  <div
    class="c1"
  >
    Погода здесь
  </div>
  <button
    class="c2"
  >
    Обновить геолокацию
  </button>
</div>
`;
